src/XrdHttp/XrdHttpReq.cc

```diff
@@ -1443,8 +1450,9 @@ int XrdHttpReq::ProcessHTTPReq() {
 
             TRACEI(REQ, "XrdHTTP PUT: Writing chunk of size " << bytes_to_write << " starting with '" << *(prot->myBuffStart) << "'" << " with " << chunk_bytes_remaining << " bytes remaining in the chunk");
             if (!prot->Bridge->Run((char *) &xrdreq, prot->myBuffStart, bytes_to_write)) {
-              prot->SendSimpleResp(500, NULL, NULL, (char *) "Could not run write request.", 0, false);
-              return -1;
+              TRACEI(REQ, " Failed to run write request on the bridge.");
```

Comment:
Maybe you can put the error message in a string allowing you to not repeat the same string in the two calls?

Reply:
Done in the latest commit.